lib/index.js

```diff
@@ -432,6 +432,71 @@ api.setRoles = function(actor, identity, callback) {
 };
 
 /**
+ * Updates an Identity's ResourceRole.
+ *
+ * @param actor the Identity performing the action.
+ * @param options the options to use:
+ *   // FIXME: update doc
+ * @param callback(err) called once the operation completes.
+ */
+api.updateRole = function(actor, options, callback) {
+  if(!options || typeof options !== 'object') {
+    throw new TypeError('options must be specified.');
+  }
+  if(typeof options.role !== 'string') {
+    throw new TypeError('options.role must be a string.');
+  }
+  if(typeof options.identityId !== 'string') {
+    throw new TypeError('options.identityId must be a string.');
+  }
+  if(typeof options.operation !== 'string') {
+    throw new TypeError('options.operation must be a string.');
+  }
+  if(!['add', 'delete'].includes(options.operation)) {
```

Comment:
I thought of another way to implement this. We could do it as capability delegation.

So, suppose we gave regular users the capability: `PERMISSIONS.PERMISSION_DELEGATE + self`

Then, when we call `addCapability` or `removeCapability`, we:

1. Check to see if the actor has `PERMISSIONS.PERMISSION_DELEGATE` for the given `resourceId` or its `owner`.
2. Get the role they want to add to another identity and all of its permissions.
3. For each permission in the role, check to see if the actor has permission for the `resourceId` or its `owner`.
4. If the actor passed all the permission checks, then allow the actor to assign the `role+resourceId` (capability) to the identity they want to assign it to. Note: We don't check `IDENTITY_EDIT` for earlier reasons (see use case above).

There are some problems with this approach that we need to consider if we care about:

1. It makes ownership special -- we assume that any resource with an owner can be treated in this special way.
2. Revoking permissions is unclear. It may be strange to use the same delegate permission for adding and removing, so we may need to have a `PERMISSIONS.PERMISSION_REVOKE` (but for that to work properly, we really need to know who the delegator was to begin with -- which is something else to consider).